### PR TITLE
26 febe upload photos send to store in be

### LINKED ISSUE #26: FE/BE: Upload Photos & send to store in BE

### PR DESCRIPTION
**Pull request by westonio**
### Description of changes
Chore:
- Add `apollo server gem` and configuration.
- Install rails `ActiveStorage`, which updates the DB schema accordingly for attachments, blobs, and variant_records tables.
Update:
- Update all GraphQL mutation tests to include an image file
- Update the JSON and GraphQL schema to reflect changes
Test:
- Test sad path for if photo not provided for Review (use default photo instead)
- Test happy path for when a photo is attached to a Review
Feature:
- Update CreateReview GraphQL mutation to accept files
- Update the Reviews and Review(id:) GraphQL queries to return the review's photoUrl 
- Add a `photo_url` model method to create a URI that can be accessed by the front end to display the image.
- Add sad path to attach a default `Foodie Brain` image if one is not sent from the FE

### Test Suite
- [x] Are all tests within the test suite passing?
- _SImpleCov Coverage is at 98.49%_

### If you used graphiql on this branch, did you remember to comment #require 'sprockets/railtie' in the config/application.rb?
-[x]

#### Add GIF (REQUIRED)
giphy

### CODE REVIEW

File: app/graphql/mutations/create_review.rb
Comment (Gabe-Torres) [inline app/graphql/mutations/create_review.rb:1]: I appreciate the conditional implementation of a default photo if a user does not attach a photo!  Edge-casing is important here for user experience

File: app/graphql/mutations/create_review.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 class Mutations::CreateReview < Mutations::BaseMutation
   argument :name, String, required: true
-  argument :photo, String # TODO make photo required
+  argument :photo, ApolloUploadServer::Upload, required: false, default_value: nil #TODO: Update to required once FE operational
```
Comment (Gabe-Torres) [inline app/graphql/mutations/create_review.rb:3]: Is ApolloUploadServer unique to graphql? I dont have much experience with it but from my understanding, it handles file uploading.

Comment (ChrisAsercion) [reply app/graphql/mutations/create_review.rb:3]: Yes! It's specific to GraphQl

File: app/graphql/mutations/create_review.rb [hunk 2]
```diff
@@ -33,6 +33,12 @@ def resolve(name:, photo:, description:, dairy_free:, gluten_free:, halal:, kosh
       lat: lat, 
       lng: lng)
 
+      if photo
+        review.photo.attach(io: File.open(photo), filename: photo.original_filename, content_type: photo.content_type)
+      else
+        review.photo.attach(io: File.open('app/assets/images/no-image-default.jpeg'), filename: 'no-image-default.jpeg', content_type: 'image/jpeg')
+      end
```
Comment (Gabe-Torres) [inline app/graphql/mutations/create_review.rb:40]: This conditional for attaching photos is a beauty. Question: where are the helper methods coming from? (.attach, .open, .original_filename)

File: app/graphql/mutations/create_review.rb [hunk 3]
```diff
@@ -33,6 +33,12 @@ def resolve(name:, photo:, description:, dairy_free:, gluten_free:, halal:, kosh
       lat: lat, 
       lng: lng)
 
+      if photo
+        review.photo.attach(io: File.open(photo), filename: photo.original_filename, content_type: photo.content_type)
```
Comment (z-fitch) [inline app/graphql/mutations/create_review.rb:37]: Did this fix the issue with the super crazy-looking url?

File: app/graphql/mutations/create_review.rb (hunk 1 above)
Comment (z-fitch) [inline app/graphql/mutations/create_review.rb:3]: Does the default nil go with the no-image-default or is it just there as a jic

**Review by z-fitch — APPROVED** [review]
This looks awesome, left some comments/questions

File: app/graphql/mutations/create_review.rb (hunk 1 above)
Comment (ChrisAsercion) [reply app/graphql/mutations/create_review.rb:3]: I think this is related to not filling the photo section in yet, but not the no-image-default

File: db/seeds.rb [hunk 4]
```diff
@@ -1,9 +1,2 @@
...
-# The data can then be loaded with the bin/rails db:seed command (or created alongside the database with db:setup).
-#
-# Examples:
-#
-#   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
-#   Character.create(name: "Luke", movie: movies.first)
-Review.create(name: "Matcha Mochi Donut", photo: "example.png", description: "Donuts made with Mochi", dairy_free: nil, gluten_free: 0, halal: 0, kosher: 0, nut_free: 0, vegan: 0, vegetarian: 0, likes: 23, dislikes: 1, lat: "39.675827212143645", lng: "-104.8654249015717") 
-Review.create(name: "Flu Shot Pho", photo: "something.jpeg", description: "Pho with lemongrass, lime, and chili broth", dairy_free: 0, gluten_free: 0, halal: 0, kosher: 0, nut_free: 0, vegan: 0, vegetarian: 0, likes: 412, dislikes: 45, lat: "39.6880903110667", lng: "-104.94002900531989") 
+Review.create(name: "Matcha Mochi Donut", photo: Rack::Test::UploadedFile.new('spec/fixtures/images/test.jpeg'), description: "Donuts made with Mochi", dairy_free: nil, gluten_free: 0, halal: 0, kosher: 0, nut_free: 0, vegan: 0, vegetarian: 0, likes: 23, dislikes: 1, lat: "39.675827212143645", lng: "-104.8654249015717")
```
Comment (ChrisAsercion) [inline db/seeds.rb:1]: Glad we have some seed data!

**Review by ChrisAsercion — APPROVED** [review]

**Review by Gabe-Torres — APPROVED** [review]

**Pull request merged**